passes/techmap/extract_reduce.cc

```diff
@@ -52,14 +52,30 @@ struct ExtractReducePass : public Pass
 		log("        Allows matching of cells that have loads outside the chain. These cells\n");
 		log("        will be replicated and folded into the $reduce_* cell, but the original\n");
 		log("        cell will remain, driving its original loads.\n");
+		log("    -debug-verbose\n");
+		log("        Debug print internal states.\n");
 		log("\n");
 	}
 
 	inline bool IsRightType(Cell* cell, GateType gt)
 	{
 		return (cell->type == "$_AND_" && gt == GateType::And) ||
 				(cell->type == "$_OR_" && gt == GateType::Or) ||
-				(cell->type == "$_XOR_" && gt == GateType::Xor);
+				(cell->type == "$_XOR_" && gt == GateType::Xor) ||
+				(cell->type == "$and" && gt == GateType::And) ||
+				(cell->type == "$or" && gt == GateType::Or) ||
+				(cell->type == "$xor" && gt == GateType::Xor);
+	}
+
+	bool is_single_bit(Cell *cell, SigMap sigmap)
```

Comment:
I think it would be a good idea to call this function from the `IsRightType` function (rather than modifying all of the call sites). This helps prevent accidentally forgetting to call `is_single_bit` in a critical location.